test/unittests/execute_floating_point_test.cpp

```diff
@@ -781,6 +781,106 @@ TEST(execute_floating_point, f64_promote_f32)
     EXPECT_GT(FP{res4.value.f64}.nan_payload(), FP64::canon);
 }
 
+TEST(execute_floating_point, f32_demote_f64)
+{
+    /* wat2wasm
+    (func (param f64) (result f32)
+      local.get 0
+      f32.demote_f64
+    )
+    */
+    const auto wasm = from_hex("0061736d0100000001060160017c017d030201000a070105002000b60b");
+    auto instance = instantiate(parse(wasm));
+
+    constexpr double f32_max = FP32::Limits::max();
+    ASSERT_EQ(f32_max, 0x1.fffffep127);
+
+    // The "artificial" f32 range limit: the next f32 number that could be represented
+    // if the exponent had a larger range.
+    // Wasm spec Rounding section denotes this as the limit_N in the float_N function (for N=32).
+    // https://webassembly.github.io/spec/core/exec/numerics.html#rounding
+    constexpr double f32_limit = 0x1p128;  // 2**128.
+
+    // The lower boundary input value that results in the infinity. The number is midway between
+    // f32_max and f32_limit. For this value rounding prefers infinity, because f32_limit is even.
+    constexpr double lowest_to_inf = (f32_max + f32_limit) / 2;
+    ASSERT_EQ(lowest_to_inf, 0x1.ffffffp127);
```

Comment:
It doesn't really matter, but is there a reason this should assert_eq and not static_assert?

Reply:
I started with `static_assert`, then decided that maybe it should build anyway even if not true. I guess static_assert may be less confusing for readers.